### ISSUE TITLE
[BUG] nil reference on macOS

### ISSUE THREAD

**Issue by jaaqo** [AvyAddons/BlizzardBars_Mouseover#5]
### Description

Logging in is greeted with Lua Error. Mac user.

### Addon Version

v2.1.1

### World of Warcraft Flavor

Retail

### Lua Error
```
Message: Interface/AddOns/BlizzardBars_Mouseover/Core.lua:314: attempt to index local 'self' (a nil value)
Time: Mon Dec 11 14:14:46 2023
Count: 1
Stack: Interface/AddOns/BlizzardBars_Mouseover/Core.lua:314: attempt to index local 'self' (a nil value)
[string "@Interface/AddOns/BlizzardBars_Mouseover/Core.lua"]:314: in function <Interface/AddOns/BlizzardBars_Mouseover/Core.lua:313>

Locals: self = nil
event = nil
isInitialLogin = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index local 'self' (a nil value)"
addon = <table> {
 OnInit = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:457
 SetBlingRender = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:260
 GetFlyoutParent = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:88
 Dragonriding = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:313
 fades = <table> {
 }
 button_names = <table> {
 }
 CheckBypass = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:109
 CancelTimer = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:76
 ShowBars = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:278
 shortName = "BlizzardBars"
 SecureHook = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:138
 HideBars = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:288
 enabled = true
 FadeBar = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:161
 CancelAllTimers = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:82
 FadeOutBarTimer = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:207
 ApplyOnBar = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:121
 HookBar = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:228
 HandleFlyoutHide = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:353
 SetBling = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:252
 HandleFlyoutShow = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:344
 timers = <table> {
 }
 IsEventRegistered = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Environment.lua:72
 MigrateDB = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Config.lua:56
 dragonriding = false
 UnregisterAllEvents = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Environment.lua:71
 Print = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Environment.lua:27
 bar_names = <table> {
 }
 RegisterUnitEvent = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Environment.lua:68
 CreateButton = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Config.lua:85
 ToggleBars = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:297
 RegisterChatCommand = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Environment.lua:48
 db = <table> {
 }
 L = <table> {
 }
 eventFrame = BlizzardBars_MouseoverEventFrame {
 }
 CreateConfigPanel = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Config.lua:174
 CreateSlider = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Config.lua:138
 RoundToNearestPercentile = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Config.lua:119
 CreateHeader = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Config.lua:105
 buttons = <table> {
 }
 RegisterEvent = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Environment.lua:67
 ComputeValues = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Config.lua:64
 OnEnable = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:493
 OnEvent = <function> defined @Interface/AddOns/BlizzardBars_Mouseover/Core.lua:416
 HookBars = <function> def
```
### Reproduction Steps

1. Add addon
2. Close game
3. Start game

### Tested with only BlizzardBars_Mouseover

- [X] Yes

**Comment by jaaqo** [comment]
`/reload` works, only a logout triggers this behavior

**Comment by lucasvienna** [comment]
I'm investigating, thanks for the report

**Comment by lucasvienna** [comment]
It should be fixed with [a8f8246](https://github.com/AvyAddons/BlizzardBars_Mouseover/commit/a8f82466c2c8d7ae44323e0fec7a3f3884e20155). If it still persists, please reopen.
